f3ast/structure.py

```diff
@@ -140,6 +141,26 @@ def rotate(self, rotation_axis, rotation_angle):
         transf_matrix[:3, :3] = r.as_matrix()
         self.apply_transform(transf_matrix)
         self.clear_slicing()
+        
+    def mirror(self, mirror_plane='x'):
```

Comment:
This function is great to have! One suggestion I would have is to make more general in an easy way. The input should not be a string, but a normal vector to the mirror plane. So the function would be:
```
def mirror(self, normal=(1, 0, 0)):
    # make sure normal is unit vector
    unit_normal = np.array(normal) / np.linalg.norm(normal)
    # create transform matrix
    transf_matrix = np.eye(4)
    transf_matrix[:3, :3] -= 2*np.outer(unit_normal, unit_normal)
    self.apply_transform(transf_matrix)
    self.clear_slicing()
```
Then you can just put in any plane you want. E.g. if you wanted to mirror it across a 45 degree plane, you could put in: `object.mirror((1, 1, 0))`, and your xyz mirrors work normally with `object.mirror((0,1,0))` for e.g. y mirror

Reply:
very, very nice :-)